pkg/http/handler/timeout_test.go

```diff
@@ -626,6 +627,99 @@ func BenchmarkTimeoutHandler(b *testing.B) {
 	})
 }
 
+func TestTimeoutHandlerConcurrentHeaderAccess(t *testing.T) {
```

Comment:
Are we sure that this test would actually catch verify the problem?
I tried running it locally without the fix 1000 times but the test didn't fail.